CMakeLists.txt

```diff
@@ -255,6 +256,11 @@ if(HARDENING)
     try_append_cxx_flags("-fcf-protection=full" TARGET hardening)
 
     if(MINGW)
+      add_library(link_ssp INTERFACE)
+      target_link_libraries(link_ssp INTERFACE
+        $<$<BOOL:${BUILD_SHARED_LIBS}>:ssp>
```

Comment:
Can you elaborate on why this is needed (if it still is: https://github.com/bitcoin/bitcoin/pull/28461). I'm suprised this is being added at this point, rather than in the hardening PR, where we actually started using ssp?

Reply:
I haven't tested. Did it? If it doesn't fail on master with the changes in https://github.com/bitcoin/bitcoin/pull/28461, it'd also be good to know why it fails here.

Reply:
> it'd also be good to know why it fails here.

Indeed. FWIW, I have `objdump` outputs as follow:
- master + https://github.com/bitcoin/bitcoin/pull/28461:
```
 objdump -x src/.libs/libbitcoinconsensus-0.dll | grep -e "DLL Name:"
	DLL Name: ADVAPI32.dll
	DLL Name: KERNEL32.dll
	DLL Name: msvcrt.dll
```
- this branch:
```
$ objdump -x build/src/script/libbitcoinconsensus.dll | grep -e "DLL Name:"
	DLL Name: ADVAPI32.dll
	DLL Name: KERNEL32.dll
	DLL Name: msvcrt.dll
```

There are no traces of `libssp`...

Reply:
I would expect lssp to be needed everywhere other than bleeding-edge build environments. Is it possible that `bitcoind.exe` without lssp links but fails to run?

Reply:
> I would expect lssp to be needed everywhere...

FWIW, it was [introduced](https://github.com/bitcoin/bitcoin/commit/811a765bef5701b167ce51a047745140df9bd04f) with the commit message:
> mingw needs libssp for hardening with dlls 

:)

Reply:
> I haven't tested. Did it? If it doesn't fail on master with the changes in [bitcoin#28461](https://github.com/bitcoin/bitcoin/pull/28461), it'd also be good to know why it fails here.

Resolved in the recent push.